pkg/token/execCredentialWriter.go

```diff
@@ -20,20 +39,67 @@ type execCredentialWriter struct{}
 
 // Write writes the ExecCredential to standard output for kubectl.
 func (*execCredentialWriter) Write(token adal.Token) error {
-	t := v1.NewTime(token.Expires())
-	ec := &v1beta1.ExecCredential{
-		TypeMeta: v1.TypeMeta{
-			APIVersion: "client.authentication.k8s.io/v1beta1",
-			Kind:       "ExecCredential",
-		},
-		Status: &v1beta1.ExecCredentialStatus{
-			Token:               token.AccessToken,
-			ExpirationTimestamp: &t,
-		},
+	fmt.Fprintln(os.Stderr, os.Getenv(execInfoEnv))
+	//os.Setenv(execInfoEnv, "TEST")
+	//fmt.Println("KUBERNETES_EXEC_INFO:", os.Getenv(execInfoEnv))
+	apiVersionFromEnv, err := helperGetApiVersionFromEnv()
+	if err != nil {
+		return err
 	}
+
+	var ec interface{}
+	t := metav1.NewTime(token.Expires())
+	if apiVersionFromEnv == "client.authentication.k8s.io/v1beta1" {
```

Comment:
use a switch to determine the api version. the rest of the code can be outside the switch block to avoid duplication

Reply:
Done already. Thanks!